### DISCUSSION TITLE
new in axis - need help

### DISCUSSION THREAD

**Discussion by ashK7** [axmolengine/axmol#766] [General]
hi, i am a cocos2dx game developer and want to start axis so kindly need to know which is the stable release available of axis. because as i have seen axis is updating everyday so how i can merge that all new changes if i have already started project.

thanks.

**Comment by halx99** [comment]
You can use the branch `dev`

**Comment by ashK7** [comment]
but that is also updated daily basis or weekly. if i have started and after a some days i will get update from you so  should i have to again update the library right ? thanks @halx99

**Reply by DelinWorks** [reply]
fork the github repository and clone it into your local machine using programs like `git or github desktop` and then make your own branch in your own repository that's based on dev or release and when it comes time to update just fetch upstream into your branch and then pull the changes into your local machine, and your project will automatically adapt and recompile the engine, unless the changes made are project breaking `like editing a cmake file which is not a common change` because you would then need to re-create the project, it's quite easy re-creating your project if it has been orginized before.

**Comment by ashK7** [comment]
okay understood. thanks @DelinWorks also you can follow the setup like cocos2dx. release version after having all bug sorted and give it version code(sequence) you can upgrade version code every 3-6 month.

**Reply by DelinWorks** [reply]
that's a good idea! what do you think? @halx99 

let's make a branch for `1.x` and then make changes under it, like 1.0.1 if it's a small change like bug fixes and 1.1.0 if it's a big change and then we keep doing this until the version major is complete with enough features, we would then create another branch called `2.x` and so on!

**Reply by aismann** [reply]
We should have a 1.0 first.
